src/processor/operator/persistent/reader/csv/driver.cpp

```diff
@@ -204,6 +204,52 @@ static void castStringToList(const char* input, uint64_t len, ValueVector* vecto
     }
 }
 
+static void validateNumElementsInList(uint64_t numElementsRead, const LogicalType& type) {
+    auto numElementsInList = FixedListType::getNumElementsInList(&type);
+    if (numElementsRead != numElementsInList) {
+        throw CopyException(StringUtils::string_format(
+            "Each fixed list should have fixed number of elements. Expected: {}, Actual: {}.",
+            numElementsInList, numElementsRead));
+    }
+}
+
+static void castStringToFixedList(const char* input, uint64_t len, ValueVector* vector,
+    uint64_t rowToAdd, const CSVReaderConfig& csvReaderConfig) {
+    // assertions
+    assert(vector->dataType.getLogicalTypeID() == LogicalTypeID::FIXED_LIST);
+    auto childDataType = FixedListType::getChildType(&vector->dataType);
+    switch (childDataType->getLogicalTypeID()) {
+    case LogicalTypeID::INT64:
+    case LogicalTypeID::INT32:
+    case LogicalTypeID::INT16:
+    case LogicalTypeID::INT8:
+    case LogicalTypeID::UINT64:
```

Comment:
I think we only support fixedlist of int64,int32,int16, can you confirm with @Ashleyhx

Reply:
yes i think so